ddtrace/contrib/django/patch.py

```diff
@@ -189,12 +189,16 @@ def traced_cache(django, pin, func, instance, args, kwargs):
                 db.ROWCOUNT, sum(1 for doc in result if doc) if result and isinstance(result, Iterable) else 0
             )
         elif command_name == "get":
-            # if valid result and check for special case for Django~3.0 that returns an empty Sentinel object as
-            # missing key
-            if result is not None and result != getattr(instance, "_missing_key", None):
-                span.set_metric(db.ROWCOUNT, 1)
-            # else result is invalid or None, set row count to 0
-            else:
+            try:
+                if result is None:
+                    span.set_metric(db.ROWCOUNT, 0)
+                # check also for special case for Django~3.0 that returns an empty Sentinel object for empty results
+                # also check if result is Iterable first since some iterables return ambiguous truth results with ``==``
```

Comment:
Ah so with `==` we get the same behaviour as `bool`? Do we know if `getattr(instance, "_missing_key")` returns a constant sentinel object, much like `None`? If so then we can just use `is` like we do with `None`.

Reply:
Yes, the `==` was resulting in the same error as `bool`, resulting in an ambiguity error when being compared to some Iterable objects such as a Pandas.Series or Pandas.DataFrame. And I don't think the `getattr(instance, "_missing_key")` returns a sentinel object, [here is where it is instantiated](https://github.com/django/django/blob/fc42edd2e63e89a80e7ca81486291f74359ef8be/django/core/cache/backends/base.py#L55). Instead, I just changed this line to check if the result is an Iterable, and if it is NOT, compare it against the `_missing_key` attribute if found.